packages/polymath-ui/src/components/EthNetworkWrapper/actions.js

```diff
@@ -115,7 +115,11 @@ export const init = (networks: Array<string>) => async (dispatch: Function) => {
   });
 
   // Check if dapp is authorized by Metamask/Mist
-  if (newProviderInjected && window.ethereum._metamask.isApproved) {
+  if (
+    !accounts.length &&
+    newProviderInjected &&
+    window.ethereum._metamask.isApproved
```

Comment:
We might also want to check for the existence of `_metamask` since that object is non standard. This unless we are sure we are using Metamask in this part of the code